src/app/components/MainContent/index.jsx

```diff
@@ -31,7 +33,7 @@ const MainContent = ({ data }) => {
 MainContent.propTypes = {
   data: shape({
     model: shape({
-      blocks: arrayOf(any),
+      blocks: arrayOf(shape(headlineBlockPropTypes), shape(textBlockPropTypes)),
```

Comment:
`headlineBlockPropTypes` and `textBlockPropTypes` don't need to be wrapped in `shape`s.

Further, you should use the `blocksWithTypes` helper in `general` propTypes.